### PR TITLE
Move sshd_config step first in boot script, add basic logging

### PR DESCRIPTION
**Pull request by cadenmarchese**
### Which issue this PR addresses:

When the scripts fail to execute fully on prod deploy, we can't get in to debug them since the SSH auth config step is at the end of the script. Additionally, if any individual step is failing, we don't have visibility on which one.

### What this PR does / why we need it:

- Moves ssh step of script to the very beginning
- Adds some basic logging so we know what step (if any) is being executed during a failure of one step

### CODE REVIEW

**Review by cblecker — APPROVED** [review]

File: pkg/deploy/generator/scripts/gatewayVMSS.sh [hunk 1]
```diff
@@ -1,12 +1,19 @@
+# We need to manually set PasswordAuthentication to true in order for the VMSS Access JIT to work
+echo "setting ssh password authentication"
+sed -i 's/PasswordAuthentication no/PasswordAuthentication yes/g' /etc/ssh/sshd_config
+
```
Comment (darthhexx) [inline pkg/deploy/generator/scripts/gatewayVMSS.sh:4]: We need to reload the SSH config or this won't take effect until the reboot.

File: pkg/deploy/generator/scripts/rpVMSS.sh [hunk 2]
```diff
@@ -1,12 +1,19 @@
+# We need to manually set PasswordAuthentication to true in order for the VMSS Access JIT to work
+echo "setting ssh password authentication"
+sed -i 's/PasswordAuthentication no/PasswordAuthentication yes/g' /etc/ssh/sshd_config
+
```
Comment (darthhexx) [inline pkg/deploy/generator/scripts/rpVMSS.sh:4]: Here as well.

**Review by darthhexx — CHANGES_REQUESTED** [review]

File: pkg/deploy/generator/scripts/gatewayVMSS.sh (hunk 1 above)
Comment (cadenmarchese) [reply pkg/deploy/generator/scripts/gatewayVMSS.sh:5]: Indeed, thanks!

**Review by darthhexx — DISMISSED** [review]
lgtm

File: pkg/deploy/generator/scripts/gatewayVMSS.sh [hunk 3]
```diff
@@ -1,12 +1,20 @@
+# We need to manually set PasswordAuthentication to true in order for the VMSS Access JIT to work
+echo "setting ssh password authentication"
+sed -i 's/PasswordAuthentication no/PasswordAuthentication yes/g' /etc/ssh/sshd_config
+systemctl restart sshd.service
```
Comment (s-amann) [inline pkg/deploy/generator/scripts/gatewayVMSS.sh:4]: Do we need to restart and could we get away with `reload` instead?

**Review by s-amann — APPROVED** [review]

File: pkg/deploy/generator/scripts/gatewayVMSS.sh (hunk 3 above)
Comment (cadenmarchese) [reply pkg/deploy/generator/scripts/gatewayVMSS.sh:4]: We can move to `reload` if it's preferred, for our case either should work (I just had more confidence in the restart)

**Review by s-amann — DISMISSED** [review]

**Review by s-amann — APPROVED** [review]

**Review by darthhexx — APPROVED** [review]

**Pull request merged**